### ISSUE TITLE
First token from macro_rules invocation bleeds token type into the macro braces

### ISSUE THREAD

**Issue by RocketRace** [rust-analyzer/rust-analyzer#9190]
This is an issue with token colorization & semantic token type (purely visual). I am using rust-analyzer in Visual Studio Code.

The braces of a `macro_rules!` macro invocation (e.g. the parentheses/brackets/braces in `foo!()`/`foo![]`/`foo!{}`) can sometimes gain the semantic token type of the first token in the body block of a `macro_rules!` macro. This can result in the braces being colored incorrectly.

Example code: 

```rust
macro_rules! foo {
    () => {
        ""
    };
}

pub fn test() {
    foo!();
}
```

The `()` in the `foo!()` macro are colored green. You can see the token type is reported as a string.

<img width="526" alt="image" src="https://user-images.githubusercontent.com/35444188/121353715-1ff79380-c969-11eb-8fc4-50dd23fa3e54.png">

Some other examples demonstrating different tokens and different braces:

```rust
macro_rules! bar {
    () => {
        0
    };
}

macro_rules! baz {
    () => {
        &0
    };
}

pub fn test() {
    bar![];
    baz!{};
}
```

<img width="541" alt="image" src="https://user-images.githubusercontent.com/35444188/121353867-4a495100-c969-11eb-902a-4f1bd0a9e79a.png">

<img width="549" alt="image" src="https://user-images.githubusercontent.com/35444188/121353891-503f3200-c969-11eb-8965-4bbd3e5caeb5.png">
